### PR TITLE
Main Menu, End Screen, DoF

### PR DESCRIPTION
**Pull request by EnigmaticDragon**
I decided to use the main menu from my game from last semester. It was more convenient for me, since it already uses TextMesh Pro, has a option menu with options in it, and also a small slide out animation. The end screen displays "Victory" or "Defeat", the game stats, and a button to return to the main menu. The screen is shown after the boss death animation, or if all bots are dead. Additionally, escape can be hit now at any time during the game to open a menu with a button to return to the main menu. Also, DoF is disabled.

### CODE REVIEW

File: Assets/Scripts/Gamemaster.cs [hunk 1]
```diff
@@ -65,7 +45,16 @@ public void Register(AIMBotController bot)
             else
             {
                 registeredBots.Add(bot);
-                bot.BotDied += CheckForLose;
+                aliveBots++;
```
Comment (nfetissow) [inline Assets/Scripts/Gamemaster.cs]: What if we register dead bots? 😱

File: Assets/Scripts/UserInterface/GameDataDisplayer.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+{
+    public class GameDataDisplayer : MonoBehaviour
+    {
+        [SerializeField] private TextMeshProUGUI txtResult;
+        [SerializeField] private TextMeshProUGUI txtTime, txtBots, txtHealth;
+
+        private void Awake()
+        {
+
+            if (!txtResult)
```
Comment (nfetissow) [inline Assets/Scripts/UserInterface/GameDataDisplayer.cs]: I didn't know this check exists :) 
Does it check for null, or?

**Comment by nfetissow** [comment]
- Fixed the sound slider not showing the actual volume value at the start of the scene

- Fixed the bot's behaviour: the destination target was acting weird and the bots were walking in the wall too much. Now they walk around more nicely.

- I also added the background music.

File: Assets/Scripts/UserInterface/GameDataDisplayer.cs (hunk 2 above)
Comment (EnigmaticDragon) [reply Assets/Scripts/UserInterface/GameDataDisplayer.cs]: Yes, that's an override in UnityEngine.Object that checks for null. ^^

**Comment by EnigmaticDragon** [comment]
Thanks for the fixes, and nice music!

**Pull request merged**